src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -320,15 +328,18 @@ public DescriptorImpl() {
         @Override
         @POST
         public FormValidation doCheckCredentialsId(@QueryParameter String credentialsId) {
-            Jenkins.get().checkPermission(CONFIGURE);
```

Comment:
Not sure how these checks are still present. For those curious, I do not believe their removal does not impact security. To hit these endpoints, you would need to make a stapler call that looks like `http://myjenkinsinstance.com/jobs/jobname/configure`. The `/job` section of the URL is invoking `Jenkins.getItem` which includes an `Item.READ` security check (in the Jenkins class). I'm less certain about `/configure/` but I believe the JobProperty class is handling permissions in the `decorateACL` method (for example, `BranchJobProperty` requires `Item.CONFIGURE`, for config, but also grants delete for dead branches, extended read for the snippet generator, etc.)

Reply:
If we feel uncomfortable with removing the checks entirely, we can add an `@AncestorInPath Item context`, and replace these with `context.checkPermissions` checks instead. If I understand correctly, these will be innocuous (where `Jenkins.checkPermissions` are not).